pinc/dpsession_via_php_sessions.inc

```diff
@@ -134,34 +138,38 @@ function mysql_session_select($sid)
 function mysql_session_write($sid, $value)
 {
     $expiration = time() + ini_get("session.cookie_lifetime");
-    $result = do_session_query("SELECT COUNT(*) FROM sessions WHERE sid = '$sid'");
-    $row = mysqli_fetch_row($result);
-    $SessionExists = $row[0];
+    $result = do_session_query(sprintf(
+        "SELECT COUNT(*) FROM sessions WHERE sid = '%s'",
+        DPDatabase::escape($sid)
+    ));
+    list($SessionExists) = mysqli_fetch_row($result);
     // The table has a uniqueness constraint on 'sid',
     // so $SessionExists must be 0 or 1.
 
-    // $value is a string (constructed by PHP's built-in session
-    // code) that might contain apostrophes (or backslashes?),
-    // which will screw up the SQL queries unless we escape them.
-    $value = mysqli_real_escape_string(DPDatabase::get_connection(), $value);
 
     if ( $SessionExists == 0 )
     {
-        $query = "
+        $query = sprintf("
             INSERT INTO sessions
                 (sid, expiration, value)
             VALUES
-                ('$sid', '$expiration', '$value')
-            ";
+                ('%s', '%s', '%s')",
+            DPDatabase::escape($sid),
+            $expiration,
```

Comment:
do we trust expiration unconditionally?

Reply:
No, per my prior code review this needs `%d` in the `sprintf()`.

Reply:
@bpfoley - looks like you resolved this but I don't see it updated, did you miss a push on your branch?